shell/platform/fuchsia/flutter/tests/flatland_platform_view_unittest.cc

```diff
@@ -0,0 +1,1358 @@
+    EXPECT_EQ(vrf.times_watched, i + 1);
+
+    // Dispatch the next vrf event.
+    vrf.ScheduleCallback(vrf_states[i]);
+    RunLoopUntilIdle();
+
+    // Make sure View.focus.getCurrent completes with the current focus state.
+    auto response3 = FakePlatformMessageResponse::Create();
+    base_view->HandlePlatformMessage(response3->WithMessage(
+        "flutter/platform_views", "{\"method\":\"View.focus.getCurrent\"}"));
+    // Duplicate View.focus.getCurrent are allowed.
+    auto response4 = FakePlatformMessageResponse::Create();
+    base_view->HandlePlatformMessage(response4->WithMessage(
+        "flutter/platform_views", "{\"method\":\"View.focus.getCurrent\"}"));
+
+    // Run event loop and check our results.
+    RunLoopUntilIdle();
+    response1->ExpectCompleted(vrf_states[i] ? "[true]" : "[false]");
+    response2->ExpectCompleted("[null]");
+    response3->ExpectCompleted(vrf_states[i] ? "[true]" : "[false]");
+    response4->ExpectCompleted(vrf_states[i] ? "[true]" : "[false]");
+  }
+}
+
+// This test makes sure that the PlatformView forwards messages on the
+// "flutter/platform_views" channel for View.focus.request.
+TEST_F(FlatlandPlatformViewTests, RequestFocusTest) {
+  MockPlatformViewDelegate delegate;
+  flutter::TaskRunners task_runners =
+      flutter::TaskRunners("test_runners",  // label
+                           flutter_runner::CreateFMLTaskRunner(
+                               async_get_default_dispatcher()),  // platform
+                           nullptr,                              // raster
+                           nullptr,                              // ui
+                           nullptr                               // io
+      );
+
+  FakeFocuser focuser;
+  fidl::BindingSet<fuchsia::ui::views::Focuser> focuser_bindings;
+  auto focuser_handle = focuser_bindings.AddBinding(&focuser);
+
+  bool create_view_called = false;
+  auto on_create_view =
+      [&create_view_called, this](
+          int64_t view_id, flutter_runner::ViewCallback on_view_created,
+          flutter_runner::FlatlandViewCreatedCallback on_view_bound,
+          bool hit_testable, bool focusable) {
+        create_view_called = true;
+        on_view_created();
+        fuchsia::ui::composition::ContentId content_id;
+        on_view_bound(std::move(content_id), MakeChildViewWatcher());
+      };
+
+  auto platform_view = PlatformViewBuilder(delegate, std::move(task_runners))
+                           .SetFocuser(std::move(focuser_handle))
+                           .SetCreateViewCallback(on_create_view)
+                           .Build();
+
+  // Cast platform_view to its base view so we can have access to the public
+  // "HandlePlatformMessage" function.
+  auto base_view = static_cast<flutter::PlatformView*>(&platform_view);
+  EXPECT_TRUE(base_view);
+
+  uint64_t view_id = 42;
+
+  std::ostringstream create_message;
+  create_message << "{"
+                 << "    \"method\":\"View.create\","
```

Comment:
Briefly comment on why this test uses a View.create message, but the test below doesn't? I'm sure you have a good reason, but it'd be nice to share.